MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -1191,45 +1187,52 @@ private void loadGMToolsForPerson(Person person) {
         JMenuHelpers.addMenuIfNonEmpty(popup, new AssignPersonToUnitMenu(gui.getCampaign(), selected));
 
         if (oneSelected && person.getStatus().isActive()) {
-            if (gui.getCampaign().getCampaignOptions().useManualMarriages()
-                    && person.oldEnoughToMarry(gui.getCampaign()) && !person.getGenealogy().hasSpouse()) {
+            if (gui.getCampaign().getCampaignOptions().isUseManualMarriages()
+                    && (gui.getCampaign().getMarriage().canMarry(gui.getCampaign(),
+                            gui.getCampaign().getLocalDate(), person, false) == null)) {
```

Comment:
The canMarry() is counterintuitive (I write this because I now thought so for the second time). It reads like a boolean. Even if its visibly not with the "== null" it still feels turned around with null signalling that canMarry is true. Ideally canMarry would be boolean and marriageBlockingReason() or something could return that string when its necessary.

Reply:
That's very fair, and... I'm not certain about it either. I made the decision for this design so that a case from the first is never missed on the second... but it's still hard to decide. I'm thinking of keeping this until random death is merged too, then checking the performance for this method versus the boolean method on a huge campaign.